src/tests/workflows/recipes/components/PreferenceRolloutFields.test.js

```diff
@@ -0,0 +1,122 @@
+import { Map, List } from 'immutable';
+import PreferenceRolloutFields, {
+  RowField,
+  serializePreferenceRows,
+  deserializePreferenceRows,
+} from 'console/workflows/recipes/components/PreferenceRolloutFields';
+
+describe('<PreferenceRolloutFields>', () => {
+  it('should work', () => {
+    const props = {
+      disabled: false,
+      form: {},
+      recipeArguments: new Map(),
+    };
+    const wrapper = () => shallow(<PreferenceRolloutFields {...props} />);
+    expect(wrapper).not.toThrow();
+    const rendered = wrapper();
+    expect(rendered).toMatchSnapshot();
+  });
+
+  describe('<RowField>', () => {
+    const props = {
+      row: {
+        name: 'name',
+        type: 'string',
+        value: 'Peter',
+      },
+    };
+
+    it('should work', () => {
+      const wrapper = () => shallow(<RowField {...props} />);
+      expect(wrapper).not.toThrow();
+      const rendered = wrapper();
+      expect(rendered).toMatchSnapshot();
+    });
+
+    // it('should be possible to remove a row', () => {
+    //   const props = {
+    //     row: {
+    //       name: 'name',
+    //       type: 'string',
+    //       value: 'Peter',
+    //     },
+    //   };
+
+    //   // const wrapper = mount(<RowField {...props} />);
```

Comment:
Here's a terrible start to how I tried to render the `RowField` component. My goal was to simulate input and simulate clicks on buttons and stuff. Could do with some help on how to mock `FormItem`.